client/src/utils/Contexts/AuthContext.ts

```diff
@@ -0,0 +1,47 @@
+import { useContext, useState, createContext } from 'react';
+import {
+  STORAGE_CREDENTIALS,
+  STORAGE_USERNAME,
+  fetcher,
+  IAuthContext,
+  APIManager,
+} from '@/utils';
+
+export const AuthContext = createContext<IAuthContext | null>(null);
+
+export const useAuthContext = () => {
+  const context = useContext(AuthContext);
+  if (!context) {
+    throw new Error('useAuthContext must be used within AuthContext.Provider');
+  }
+  return context;
+};
+
+export const useAuthContextValues = () => {
+  const [credentials, updateCredentials] = useState<string | null>(() =>
+    localStorage.getItem(STORAGE_CREDENTIALS),
+  );
+  const [username, updateUsername] = useState<string | null>(() =>
+    localStorage.getItem(STORAGE_USERNAME),
+  );
+
+  const createCredentials = async (username: string, password: string) => {
+    const encodedCredentials = window.btoa(decodeURI(`${username}:${password}`));
+    await fetcher(APIManager.validateUserURL, {
+      headers: { Authorization: `Basic ${encodedCredentials}` },
+    });
+    localStorage.setItem(STORAGE_CREDENTIALS, encodedCredentials);
+    localStorage.setItem(STORAGE_USERNAME, username);
+    updateCredentials(encodedCredentials);
+    updateUsername(username);
+  };
+
+  const deleteCredentials = () => {
```

Comment:
To trzeba zrobić na porządnie. Ale to była moja decyzja żeby użyć http Basic auth i będę z nią żył. Ale dług techniczny trzeba spłacić, i mam nadzieje że ten konkretny spłacimy do stycznia 2024